src/displayapp/screens/Navigation.cpp

```diff
@@ -23,105 +23,123 @@
-    {"invalid-slight-left", "\xEE\xA4\x87"},
-    {"invalid-slight-right", "\xEE\xA4\x88"},
-    {"invalid-straight", "\xEE\xA4\x84"},
-    {"invalid-uturn", "\xEE\xA4\x89"},
-    {"merge-left", "\xEE\xA4\x97"},
-    {"merge-right", "\xEE\xA4\x98"},
-    {"merge-slight-left", "\xEE\xA4\x99"},
-    {"merge-slight-right", "\xEE\xA4\x9A"},
-    {"merge-straight", "\xEE\xA4\x84"},
-    {"new-name-left", "\xEE\xA4\x85"},
-    {"new-name-right", "\xEE\xA4\x86"},
-    {"new-name-sharp-left", "\xEE\xA4\x9B"},
-    {"new-name-sharp-right", "\xEE\xA4\x9C"},
-    {"new-name-slight-left", "\xEE\xA4\x87"},
-    {"new-name-slight-right", "\xEE\xA4\x88"},
-    {"new-name-straight", "\xEE\xA4\x84"},
-    {"notification-left", "\xEE\xA4\x85"},
-    {"notification-right", "\xEE\xA4\x86"},
-    {"notification-sharp-left", "\xEE\xA4\x9B"},
-    {"notification-sharp-right", "\xEE\xA4\xA5"},
-    {"notification-slight-left", "\xEE\xA4\x87"},
-    {"notification-slight-right", "\xEE\xA4\x88"},
-    {"notification-straight", "\xEE\xA4\x84"},
-    {"off-ramp-left", "\xEE\xA4\x9D"},
-    {"off-ramp-right", "\xEE\xA4\x9E"},
-    {"off-ramp-slight-left", "\xEE\xA4\x9F"},
-    {"off-ramp-slight-right", "\xEE\xA4\xA0"},
-    {"on-ramp-left", "\xEE\xA4\x85"},
-    {"on-ramp-right", "\xEE\xA4\x86"},
-    {"on-ramp-sharp-left", "\xEE\xA4\x9B"},
-    {"on-ramp-sharp-right", "\xEE\xA4\xA5"},
-    {"on-ramp-slight-left", "\xEE\xA4\x87"},
-    {"on-ramp-slight-right", "\xEE\xA4\x88"},
-    {"on-ramp-straight", "\xEE\xA4\x84"},
-    {"rotary", "\xEE\xA4\xA1"},
-    {"rotary-left", "\xEE\xA4\xA2"},
-    {"rotary-right", "\xEE\xA4\xA3"},
-    {"rotary-sharp-left", "\xEE\xA4\xA4"},
-    {"rotary-sharp-right", "\xEE\xA4\xA5"},
-    {"rotary-slight-left", "\xEE\xA4\xA6"},
-    {"rotary-slight-right", "\xEE\xA4\xA7"},
-    {"rotary-straight", "\xEE\xA4\xA8"},
-    {"roundabout", "\xEE\xA4\xA1"},
-    {"roundabout-left", "\xEE\xA4\xA2"},
-    {"roundabout-right", "\xEE\xA4\xA3"},
-    {"roundabout-sharp-left", "\xEE\xA4\xA4"},
-    {"roundabout-sharp-right", "\xEE\xA4\xA5"},
-    {"roundabout-slight-left", "\xEE\xA4\xA6"},
-    {"roundabout-slight-right", "\xEE\xA4\xA7"},
-    {"roundabout-straight", "\xEE\xA4\xA8"},
-    {"turn-left", "\xEE\xA4\x85"},
-    {"turn-right", "\xEE\xA4\x86"},
-    {"turn-sharp-left", "\xEE\xA4\x9B"},
-    {"turn-sharp-right", "\xEE\xA4\xA5"},
-    {"turn-slight-left", "\xEE\xA4\x87"},
-    {"turn-slight-right", "\xEE\xA4\x88"},
-    {"turn-straight", "\xEE\xA4\x84"},
-    {"updown", "\xEE\xA4\xA9"},
-    {"uturn", "\xEE\xA4\x89"},
+  struct Icon {
+    const char* fileName;
+    int16_t offset;
+  };
+
+  constexpr uint16_t iconHeight = -80;
+  constexpr uint8_t flagIndex = 18;
+  constexpr uint8_t maxIconsPerFile = 25;
+  const char* iconsFile0 = "F:/images/navigation0.bin";
+  const char* iconsFile1 = "F:/images/navigation1.bin";
+
+  constexpr std::array<std::pair<const char*, uint8_t>, 86> iconMap = {{
+    {"arrive-left", 1},
```

Comment:
It would be good if we could get rid of these magic numbers. Either by sorting by the index, or using maths to calculate the index (not sure if that's possible).

Reply:
>  It doesn't look to me like you're checking that the image files exist in the filesystem before using them

Good point, I definitely need to add this!

> It would be good if we could get rid of these magic numbers

Those indexes heavily depends on how the picture was generated (the index correspond to the position of the icon in the whole picture, knowing that each icon has a height of 80px). I don't have any ideas to improve this, except maybe add a comment to explain this.

Reply:
Yeah, it doesn't seem like much can be done about it other than adding a comment.

Reply:
I've just added a comment about the layout of the pictures and the indexing of the icons.

About checking for the file availability : LVGL will gracefully handle the case where the file is not available (no error, no crash, simply displays nothing). So the application will work but won't display the icon.
I'll however work on something similar than what we did for the watchfaces so that user won't be able to run the app if the files are not available. Since this change will impact other parts of the firmware, I'll do this in another PR in the next few days :)